tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/swtbot/PitResultNotifier.java

```diff
@@ -16,36 +16,97 @@
 
 package org.pitest.pitclipse.ui.swtbot;
 
+import java.io.IOException;
+
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.swt.finder.SWTBot;
+import org.eclipse.swtbot.swt.finder.utils.SWTBotPreferences;
+import org.eclipse.swtbot.swt.finder.waits.ICondition;
 import org.pitest.pitclipse.core.extension.point.ResultNotifier;
-import org.pitest.pitclipse.ui.extension.point.PitUiUpdate;
+import org.pitest.pitclipse.runner.PitResults;
 import org.pitest.pitclipse.ui.swtbot.ResultsParser.Summary;
 
-public class PitResultNotifier implements ResultNotifier<PitUiUpdate> {
-    @Override
-    public void handleResults(PitUiUpdate updateEvent) {
-        notifiyTestsOfHtmlResults(updateEvent);
-    }
+public class PitResultNotifier implements ResultNotifier<PitResults> {
+    public enum PitSummary {
+        INSTANCE;
 
-    private void notifiyTestsOfHtmlResults(PitUiUpdate updateEvent) {
-        PitResultsView view = buildResultsView(updateEvent);
-        tryNotifyResults(view);
-    }
+        private Summary summary;
+        private SWTWorkbenchBot bot;
+
+        public int getClasses() {
+            bot.waitUntil(new ICondition() {
+                public boolean test() throws Exception {
+                    return PitSummary.INSTANCE.getSummary() != null;
+                }
+
+                public void init(SWTBot bot) {
+                    // Intentional empty
+                }
+
+                public String getFailureMessage() {
+                    return "No summary set.";
+                }
+            }, SWTBotPreferences.TIMEOUT);
+            return summary.getClasses();
+        }
+
+        public double getCodeCoverage() {
+            bot.waitUntil(new ICondition() {
+                public boolean test() throws Exception {
+                    return PitSummary.INSTANCE.getSummary() != null;
+                }
+
+                public void init(SWTBot bot) {
+                    // Intentional empty
+                }
+
+                public String getFailureMessage() {
+                    return "No summary set.";
+                }
+            }, SWTBotPreferences.TIMEOUT);
+            return summary.getCodeCoverage();
+        }
 
-    private PitResultsView buildResultsView(PitUiUpdate results) {
-        ResultsParser parser = new ResultsParser(results.getHtml());
-        Summary summary = parser.getSummary();
-        PitResultsView view = PitResultsView.builder().withClassesTested(summary.getClasses())
-                .withTotalCoverage(summary.getCodeCoverage()).withMutationCoverage(summary.getMutationCoverage())
-                .build();
-        return view;
+        public double getMutationCoverage() {
+            bot.waitUntil(new ICondition() {
```

Comment:
Replace `ICondition` with
`org.eclipse.swtbot.swt.finder.waits.DefaultCondition`. You won't have to implement `init` anymore.